digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/MunicipalityApi.scala

```diff
@@ -1,1132 +1,390 @@
-          if (!Seq(30, 40, 50, 60, 70, 80, 90, 100, 120).contains(value))
-            halt(BadRequest(s"The property values for the property with name speed limit are not valid."))
-        }
-        case Obstacles.typeId => extractPropertyValue(getAssetNameProp(assetTypeId).head, prop, propertyValueToInt).asInstanceOf[Seq[Int]].foreach {
-          value => if (!Seq(1,2).contains(value))
-            halt(BadRequest(s"The property values for the property with name obstacleType are not valid."))
-        }
-        case PedestrianCrossings.typeId  => extractPropertyValue(getAssetNameProp(assetTypeId).head, prop, propertyValueToInt).asInstanceOf[Seq[Int]].foreach { value =>
-          if (!Seq(0,1).contains(value))
-            halt(BadRequest(s"The property values for the property with name hasPedestrianCrossing are not valid."))
-        }
-        case RailwayCrossings.typeId  =>
-            extractPropertyValue("safetyEquipment", prop, propertyValueToInt).asInstanceOf[Seq[Int]].foreach { value =>
-            if (!Seq(1,2,3,4,5).contains(value))
-              halt(BadRequest(s"The property values for the property with name safetyEquipment is not valid."))
-          }
-        case TrafficLights.typeId  => extractPropertyValue(getAssetNameProp(assetTypeId).head, prop, propertyValueToInt).asInstanceOf[Seq[Int]].foreach { value =>
-          if (!Seq(0,1).contains(value))
-            halt(BadRequest(s"The property values for the property with name hasTrafficLight are not valid."))
-        }
-        case assetType7restrictions if Maximum7Restrictions.contains(assetType7restrictions) =>
-          extractPropertyValue(getAssetNameProp(assetTypeId).head, prop, propertyValueToInt).asInstanceOf[Seq[Int]].foreach { value =>
-            if(!value.toString.forall(_.isDigit) || value <= 0)
-              halt(BadRequest(s"The property values for " + AssetTypeInfo.apply(assetTypeId).label + " asset type are not valid."))
-          }
-        case RoadWidth.typeId => extractPropertyValue(getAssetNameProp(assetTypeId).head, prop, propertyValueToInt).asInstanceOf[Seq[Int]].foreach { value =>
-          if(!value.toString.forall(_.isDigit) || value <= 0)
-            halt(BadRequest(s"The property values for "  + AssetTypeInfo.apply(assetTypeId).label + " asset type are not valid."))
+            featuresWithoutIds.sum
+          case None =>
+            awsDao.updateDatasetStatus(dataset.datasetId, DatasetStatus.ErrorsFeatures.value)
+            1
         }
-        case NumberOfLanes.typeId => extractPropertyValue(getAssetNameProp(assetTypeId).head, prop, propertyValueToInt).asInstanceOf[Seq[Int]].foreach { value =>
-          if(!value.toString.forall(_.isDigit) || value <= 0)
-            halt(BadRequest(s"The property values for "  + AssetTypeInfo.apply(assetTypeId).label + " asset type are not valid."))
+        missingIds
+      }).sum
+      featuresWithoutIds
+    } else {
+      0
```

Comment:
Avoid this, returns an optional value, None in case you have nothing.

Reply:
Applied this suggestion